contents/docs/contribute/developing-locally.mdx

```diff
@@ -9,264 +9,282 @@ showTitle: true
 
-First clone the repository:
+> **Another friendly tip:** You may run into `psycopg2` errors while migrating on an Apple Silicon machine. Try out the steps in this [comment](https://github.com/psycopg/psycopg2/issues/1216#issuecomment-820556849) to resolve this.
 
-```bash
-git clone https://github.com/PostHog/posthog
-```
+### 6. Start PostHog
 
-Then start the instance with `docker-compose`:
+Now start all of PostHog (backend, worker, plugin server, and frontend – simultaneously) with:
 
 ```bash
-docker-compose -f docker-compose.dev.yml up
+export KAFKA_HOSTS=kafka:9092
+./bin/start
 ```
 
-> **It may take several minutes to build the system for the first time**. If you see an error message once your app has launched saying the front-end is not built, please wait for the Yarn dependencies to be installed (the logs will output a `✔ Webpack: Compiled successfully` message once the frontend is ready).
-
-> By default, Docker images include all dependencies for full SAML support. If you want to run your Docker image without these dependencies, please set either `SAML_DISABLED=1` environment variable or `saml_disabled=1` build argument at build time.
-
-</HiddenSection>
+Open [http://localhost:8000](http://localhost:8000) to see the app.
 
-<br />
-<hr />
+> **Note:** The first time you run this command you might get an error that says "layout.html is not defined". Make sure you wait until the frontend is finished compiling and try again.
 
-## Useful commands
+To see some data on the frontend, you should go to the `http://localhost:8000/demo` and play around with it. This will give you a Hogflix test project containing some data data in the app.
 
-### Running backend, worker, and frontend all together
 
-This script runs the below three scripts concurrently, and automatically sets `DEBUG=1`.
+## Testing
 
-Run `./bin/start`
+For a PostHog PR to be merged, all tests must be green, and ideally you should be introducing new ones as well – that's why you must be able to run tests with ease.
 
-### Running backend separately (Django)
-
-Run `DEBUG=1 ./bin/start-backend`
-
-### Running background worker separately (Celery)
+For backend, simply use:
+```bash
+pytest
+```
+You can narrow the run down to only files under matching paths:
+```bash
+pytest posthog/test/test_entity_model.py
+```
+Or to only test cases with matching function names:
+```bash
+pytest posthog/test/test_entity_model.py -k test_inclusion
+```
+To see debug logs (such as ClickHouse queries), add argument `--log-cli-level=DEBUG`.
 
-Run `DEBUG=1 ./bin/start-worker`
+For Cypress end-to-end test, run `bin/e2e-test-runner`. This will temporarily install required dependencies inside the project, spin up a test instance of PostHog, and show you the Cypress interface, from which you'll manually choose tests to run.
```

Comment:
```suggestion
For Cypress end-to-end test, run `bin/e2e-test-runner`. This will temporarily install required dependencies inside the project, spin up a test instance of PostHog, and show you the Cypress interface, from which you'll manually choose tests to run.

Once you're done, terminate the command with cmd + C. Be sure to wait until the command terminates gracefully so temporary dependencies are removed from `package.json` and you don't commit them accidentally.
```

Reply:
done!